### ISSUE TITLE
GroupEx Pro Integration Refactoring 

### ISSUE THREAD

**Issue by ddrozdik** [ymcatwincities/openy#1021]
Currently, there are 5 modules that provide different features to work with GroupEx Pro. 
For example: 
- **embedded_groupexpro_schedule** - Provides a form with settings to set up what account should be used and block that will render iframe with GroupEx pro content
- **openy_prgf_embedded_groupexpro_schedule** - Provides a paragraph to embed GroupEx Pro schedule. 
- **openy_group_schedules** - Provides a full integration with GroupEx Pro using GroupEx Pro API and it **also** provides settings to set up what account should be used. 
- **groupex_form_cache** - Caching for Groupex requests. 
- **openy_digital_signage_groupex_schedule** - integrates with GroupEx Pro to get data for specific locations

The **first** one is fully independent but it provides settings to configure what GroupEx account to use. 
The **second** and the **third** modules these two modules were backporteed to Open Y from other websites and do integration with API and provide some pages, and features like caching requests. 
The **fourth** module reuses some of the features from the second module, but also provides some own features needed only for Digital Signage. 

At the moment, in case I want to use **openy_digital_signage_groupex_schedule** I can't do that without enabling **openy_group_schedules** and **groupex_form_cache**, even if I do not want to use features from those modules. 

I would like to recommend to do refactoring of the modules **openy_group_schedules** and **groupex_form_cache** to move out code needed for API requests to a separate module, move out code needed for GroupEx mapping entities to a separate module, and make all of them decoupled as much as possible, so other modules could add small modules as dependencies instead of these two big modules. 
Also, would be great to organize al modules that work with GroupEx into one module, which will provide submodules with different features, see example:
* openy_groupex
   * openy_groupex_api
   * openy_groupex_cache
   * openy_groupex_ui
   * openy_groupex_mapping
   * openy_groupex_schedules
  * embedded_groupexpro_schedule
* openy_digital_signage_groupex_schedule

cc @podarok, @YMCA-GTC , @Sanchiz

**Comment by YMCA-GTC** [comment]
@ddrozdik Will this refactoring work be handled by 5J with the Decoupling work?

**Comment by ddrozdik** [comment]
@YMCA-GTC no, it will not. This is something big and will take some time to refactor carefully. Also first of all this task require detailed architecture to avoid issues in the future.

**Comment by ddrozdik** [comment]
@YMCA-GTC we will try to include this into the Open Y 2.0 release

**Comment by podarok** [comment]
With PEF is there any demand to keep embedded GroupEx PRO integration?
YGTC team if almost done with Syncer implementation for PEF - AB test will start soon and code will be published afterward. 
ALso we found a bug in GroupEX PRO which should be fixed by GroupEx

**Comment by ddrozdik** [comment]
I think with PEF - GroupEx integration there is no need for embedded_groupexpro_schedule module. But anyway refactoring is needed. 
Also according to latest discussions of moving 3rd party integrations like ActiveNet, Daxko, GroupEx, etc to separated repositories make sense to refactor code a bit and leave only really necessary modules.

**Comment by podarok** [comment]
HI @ddrozdik 
With @danylevskyi we need to check what is left after CMS optimization project 5Jars did for YGTC and what was improved by our team during AB test of Schedules for YGTC. We should verify if "New Schedules" from YGTC backported fully or not and backport it into OpenY. Then I'll record a video tutorial how to set Syncer service for proper classes pulling from GroupEx PRO into OpenY.

Refactoring should be justified by business demand. Refactoring for the refactoring makes no sense.
Currently, we have no that mush information about the frequency of usage Groupex PRO( Daxko ) embedded widgets, this should be covered by analytics we put in OpenY recently. @paige-kiecker expects us to have at least some data end of 2020 when a majority of Ys updated to the version, which adds analytics. 
Given the fact embedded schedule probably used by small Ys - that timeframe could be extended because upgrading them could be postponed.
Can we discuss this task, created ~2 years ago on some of our Weekly calls with 5Jars and what is relevant today for the next steps to be taken?
Thank you

**Comment by sarah-halby** [comment]
More information is needed to create a business case for this item due to its size and complexity. @podarok would like to discuss this as a future addition for Activity Finder enhancements in Q2 - Q3 of 2021. @ymcagtc FYI for Activity Finder planning.
